### ISSUE TITLE
`href` is missing in the section heading

### ISSUE THREAD

**Issue by ikatyang** [vue-styleguidist/vue-styleguidist#971]
**Current behavior**

There is no href for headings, and there are warnings:

```txt
Warning: Failed prop type: The prop `href` is marked as required in `SectionHeadingRenderer`, but its value is `undefined`.
```

```html
<!-- the current output html for the heading -->
<h1 id="components" class="rsg--heading-14 rsg--heading1-15">
  <a class="rsg--sectionName-11">Components</a>
</h1>
```

**To reproduce**

```sh
mkdir no-heading-href-repro
cd no-heading-href-repro
yarn init -y
yarn add vue vue-loader vue-styleguidist vue-template-compiler webpack
echo '
const path = require("path");
const { VueLoaderPlugin } = require("vue-loader");
module.exports = {
  sections: [
    {
      name: "Components",
      components: path.resolve(__dirname, "./src/components/*.vue"),
    },
  ],
  webpackConfig: {
    module: { rules: [{ test: /\.vue$/, loader: "vue-loader" }] },
    plugins: [new VueLoaderPlugin()],
  },
};
' > styleguide.config.js
mkdir -p src/components
echo '
<template>
  <div>Foo</div>
</template>

<script>
export default {
  name: "Foo",
};
</script>
' > src/components/Foo.vue
yarn vue-styleguidist server --open
```

**Expected behavior**

There should be a href for the heading "Components".

**Environments**

- vue-styleguidist: 4.32.2

**Comment by elevatebart** [comment]
agreed, and I just fixed it

**Comment by elevatebart** [comment]
Feel free to re-open if it is still failing

**Comment by ikatyang** [comment]
Thanks for the quick fix, the warnings are fixed. But I upgraded to 4.32.3 with the above repro and it seems the `<a>` tag is replaced by `<span>` while the link style is still there, which looks weird:

```html
<h1 id="components" class="rsg--heading-14 rsg--heading1-15">
  <span class="rsg--sectionName-11">Components</span>
</h1>
```

```css
.rsg--sectionName-11:hover, .rsg--sectionName-11:active {
    text-decoration: underline;
    cursor: pointer;
}
```

And it seems the `href` should be provided by the user somehow (1aaf4ee4569848202f009e19529a541b9e4d0840) but I'm not sure why do we need to provide it given that the heading URL fragments (e.g., `#my-heading`) should be auto-generated.

**Comment by elevatebart** [comment]
I am glad it worked! 
You are right I think I should remove underline and pointer if there is no link.
I will fix this soon.

All `href` are built at the root level.
The reason why href are provided by the root components are:
- We need urls to be unique so that we can manage routing properly. If 2 components have the same name we can calculate slugs to avoid conflicts.
- A feature is in the works to be able to link one component's documentation to another. To resolve these links we need all urls to be available for every component.

I hope my answer helps

**Comment by ikatyang** [comment]
Sorry I did not provide the actual usecase, I thought the minimal repro (for the `href` issue) should be enough, but it seems the minimal repro is not suitable for the case you mentioned above. Here's the more appropriate repro modified from the initial repro:

```diff
 // styleguide.config.js
 const path = require("path");
 const { VueLoaderPlugin } = require("vue-loader");
 module.exports = {
   pagePerSection: true,
   sections: [
     {
       name: "Components",
+      sectionDepth: 1,
+      sections: [
+        {
+          name: "My Heading",
           components: path.resolve(__dirname, "./src/components/*.vue"),
+        },
+      ],
     },
   ],
   webpackConfig: {
     module: { rules: [{ test: /\.vue$/, loader: "vue-loader" }] },
     plugins: [new VueLoaderPlugin()],
   },
 };
```

If I click the "My Heading" in the sidebar, it'll navigate to the "My Heading" section with url updated (`http://localhost:6060/#/Components/My%20Heading`). I thought clicking the section heading should behave the same as clicking the corresponding link in the sidebar, which brings us to the clicked section with the url updated, but the "My Heading" section heading is now a `<span>` instead of an `<a>`, is this behavior expected?

**Comment by elevatebart** [comment]
I will definitely look into it.
